### ISSUE TITLE
CREATE EXTENSION postgis is throwing "Connection to the server has been lost." error

### ISSUE THREAD

**Issue by sandeepm212** [appropriate/docker-postgis#85]
Hi,

i am trying to add the postgis extension to the DB that is created by the dicker file.
I am getting the below error.
"Connection to the server has been lost."

Tried the execution from pgadmin and command line.

<img width="990" alt="screen shot 2018-02-04 at 12 44 41 am" src="https://user-images.githubusercontent.com/3884435/35775802-cbac67d4-0944-11e8-9c76-234dc4103a0c.png">
<img width="508" alt="screen shot 2018-02-04 at 12 45 47 am" src="https://user-images.githubusercontent.com/3884435/35775803-cbc373f2-0944-11e8-9912-dde017e76e24.png">

**Comment by ImreSamu** [comment]
>"Connection to the server has been lost."

* check your network
* check the server log
* try to connect with `psql`
* ...

> i am trying to add the postgis extension to the DB

no need,  it is created by automatically. 
* see:  https://github.com/appropriate/docker-postgis/blob/master/10-2.4/initdb-postgis.sh#L18

 https://github.com/appropriate/docker-postgis/issues/84

**Comment by sandeepm212** [comment]
Hi @ImreSamu 

I checked the server logs, i see the below error.

Could you please let me know what steps i can do to resolve this issue?

**Error:**
2018-02-04 15:33:20.825 UTC [1] LOG:  server process (PID 301) was terminated by signal 11: Segmentation fault
2018-02-04 15:33:20.825 UTC [1] DETAIL:  Failed process was running: 
	        -- Upgrade PostGIS (includes raster)
	        CREATE EXTENSION IF NOT EXISTS postgis VERSION '2.4.3';
	        ALTER EXTENSION postgis  UPDATE TO '2.4.3';
	
	        -- Upgrade Topology
	        CREATE EXTENSION IF NOT EXISTS postgis_topology VERSION '2.4.3';
	        ALTER EXTENSION postgis_topology UPDATE TO '2.4.3';
	
	        -- Upgrade US Tiger Geocoder
	        CREATE EXTENSION IF NOT EXISTS postgis_tiger_geocoder VERSION '2.4.3';
	        ALTER EXTENSION postgis_tiger_geocoder UPDATE TO '2.4.3';
	    
2018-02-04 15:33:20.825 UTC [1] LOG:  terminating any other active server processes
2018-02-04 15:33:20.825 UTC [271] WARNING:  terminating connection because of crash of another server process
2018-02-04 15:33:20.825 UTC [271] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2018-02-04 15:33:20.825 UTC [271] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2018-02-04 15:33:20.825 UTC [269] WARNING:  terminating connection because of crash of another server process
2018-02-04 15:33:20.825 UTC [269] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2018-02-04 15:33:20.825 UTC [269] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2018-02-04 15:33:20.825 UTC [266] WARNING:  terminating connection because of crash of another server process
2018-02-04 15:33:20.825 UTC [266] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2018-02-04 15:33:20.825 UTC [266] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2018-02-04 15:33:20.826 UTC [270] WARNING:  terminating connection because of crash of another server process
2018-02-04 15:33:20.826 UTC [270] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2018-02-04 15:33:20.826 UTC [270] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2018-02-04 15:33:20.837 UTC [1] LOG:  all server processes terminated; reinitializing
2018-02-04 15:33:20.850 UTC [302] LOG:  database system was interrupted; last known up at 2018-02-04 15:32:52 UTC
2018-02-04 15:33:21.008 UTC [302] LOG:  database system was not properly shut down; automatic recovery in progress
2018-02-04 15:33:21.010 UTC [302] LOG:  redo starts at 0/3B5A318
2018-02-04 15:33:21.014 UTC [302] LOG:  unexpected pageaddr 0/1C00000 in log segment 000000010000000000000003, offset 12582912
2018-02-04 15:33:21.014 UTC [302] LOG:  redo done at 0/3BFFFA0
2018-02-04 15:33:21.036 UTC [1] LOG:  database system is ready to accept connections

**Comment by sandeepm212** [comment]
Hi All,

To add more info, complete log when the docker is run for the first time.

**Full Docker Startup log:**
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data ... ok
creating subdirectories ... ok
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting dynamic shared memory implementation ... posix
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data -l logfile start


WARNING: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
waiting for server to start....2018-02-04 16:13:45.032 UTC [39] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2018-02-04 16:13:45.032 UTC [39] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2018-02-04 16:13:45.032 UTC [39] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2018-02-04 16:13:45.035 UTC [39] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2018-02-04 16:13:45.043 UTC [40] LOG:  database system was shut down at 2018-02-04 16:13:44 UTC
2018-02-04 16:13:45.047 UTC [39] LOG:  database system is ready to accept connections
 done
server started
CREATE DATABASE

CREATE ROLE


**/usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/postgis.sh**
CREATE DATABASE
UPDATE 1
Loading PostGIS extensions into template_postgis
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
connection to server was lost
2018-02-04 16:13:46.389 UTC [39] LOG:  server process (PID 82) was terminated by signal 11: Segmentation fault
2018-02-04 16:13:46.389 UTC [39] DETAIL:  Failed process was running: CREATE EXTENSION IF NOT EXISTS postgis;
2018-02-04 16:13:46.389 UTC [39] LOG:  terminating any other active server processes
2018-02-04 16:13:46.389 UTC [44] WARNING:  terminating connection because of crash of another server process
2018-02-04 16:13:46.389 UTC [44] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
2018-02-04 16:13:46.389 UTC [44] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
2018-02-04 16:13:46.391 UTC [39] LOG:  all server processes terminated; reinitializing

**Comment by sandeepm212** [comment]
Hi,

The issue was resolved after using 9.6 version. Looks like the issue exists only with the latest version.

Thanks,

**Comment by ImreSamu** [comment]
>Looks like the issue exists only with the latest version.

the latest versions is working for me.  
```
mdillon/postgis     10-alpine           17345023afc0        2 weeks ago         170MB
mdillon/postgis     10                  1d8860258b20        2 weeks ago         439MB
```
my guess:   it is a caching problem with your docker environment   
* see:  `docker volume ls` , `docker system prune`, ...
